### PR TITLE
Add __torch_function__ benchmarks.

### PR DESCRIPTION
**Pull request by hameerabbasi**
Since the last one was apparently reverted.

### CODE REVIEW

**Comment by ezyang** [comment]
Last time https://github.com/pytorch/pytorch/pull/34645

File: benchmarks/overrides_benchmark/bench.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    types = torch.Tensor, SubTensor, WithTorchFunction, SubWithTorchFunction
+
+    for t in types:
+        tensor_1 = t(1)
+        tensor_2 = t(2)
+
+        bench_min, bench_std = bench(tensor_1, tensor_2)
+        print(
+            "Type {0} had a minimum time of {1} \u03bcs"
+            " and a standard deviation of {2} \u03bcs.".format(
```
Comment (ezyang) [inline benchmarks/overrides_benchmark/bench.py]: You haven't fixed the issue that caused CI to fail?

```
Mar 26 20:13:22 + python bench.py -n 1 -m 1
Mar 26 20:13:22 ~/workspace/benchmarks/overrides_benchmark ~/workspace
Mar 26 20:13:22 Traceback (most recent call last):
Mar 26 20:13:22   File "bench.py", line 67, in <module>
Mar 26 20:13:22     main()
Mar 26 20:13:22   File "bench.py", line 61, in main
Mar 26 20:13:22     t.__name__, (10 ** 6) * bench_min, (10 ** 6) * bench_std,
Mar 26 20:13:22 UnicodeEncodeError: 'ascii' codec can't encode character '\u03bc' in position 54: ordinal not in range(128)
```

You may have to push to pytorch/pytorch on a branch named `ci-all/your-branch-name` to trigger the relevant ci job

Comment (hameerabbasi) [reply benchmarks/overrides_benchmark/bench.py]: Ah, I assumed it was hitting that error when parsing the file, not when printing the char.

Comment (hameerabbasi) [reply benchmarks/overrides_benchmark/bench.py]: > You may have to push to pytorch/pytorch on a branch named ci-all/your-branch-name to trigger the relevant ci job

Pushed to `ci-all/torch-function-benchmark`

**Comment by hameerabbasi** [comment]
This time, the failures really are unrelated.

**Comment by suo** [comment]
I think this broke master: 
```
Apr 01 15:52:49 + python python pyspybench.py Tensor -n 1
Apr 01 16:33:10 python: can't open file 'python': [Errno 2] No such file or directory
```

**Comment by hameerabbasi** [comment]
I apologize, I wonder why the CI didn't catch that.

**Pull request closed without merging**